### PR TITLE
Ensemble load fix

### PR DESCRIPTION
**Pull request by claudio525**
- Added IM lazy loading to both Ensemble and IMEnsemble

### CODE REVIEW

File: calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py [hunk 1]
```diff
@@ -147,3 +141,29 @@ def __load_rupture_df(self):
...
+        )
+
+    def __load_stations(self):
+        branch_stations = set.intersection(
+            *[set(branch.stations) for branch in self.branches_dict.values()]
+        ).intersection(self.ensemble.stations_ll_df.index)
+
+        self._stations = self.ensemble.stations_ll_df.loc[
+            branch_stations
+        ].sort_index()
```
Comment (joelridden) [inline calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py:169]: No newline EOF, maybe this file needs to be blacked?

**Review by joelridden — COMMENTED** [review]
Looks good to me

**Review by jasonmotha — DISMISSED** [review]
Pending formatting changes

File: calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py [hunk 2]
```diff
@@ -147,3 +139,27 @@ def __load_rupture_df(self):
                 self._rupture_df = self._rupture_df.loc[
                     ~self._rupture_df.index.duplicated()
                 ]
+
+    def __load_ims(self):
+        if IMType.pSA in self._im_types:
+            self._ims = set(list(self.branches[0].ims))
```
Comment (SnowNooDLe) [inline calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py:145]: Just a thought to be consistent,
Shall we have the following code instead? so we can safely say `self._ims` is always a list
```python
list(set(list(self.branches[0].ims)))
```

**Review by SnowNooDLe — COMMENTED** [review]
Just one comment wants to be checked.

and I triggered another benchmark test for this PR as well.

File: calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py (hunk 2 above)
Comment (SnowNooDLe) [reply calculation/gmhazard_calc/gmhazard_calc/gm_data/IMEnsemble.py:145]: But at the same time, the set is still iterable and we will polish this `self._ims` again to be `np.array` so probably not worth casting it.

**Review by SnowNooDLe — APPROVED** [review]

**Review by joelridden — APPROVED** [review]

**Pull request merged**